Helpers/CommonMenuItemPartMigrator.cs

```diff
@@ -0,0 +1,128 @@
+﻿using Etch.OrchardCore.Menu.Models;
+using Etch.OrchardCore.Widgets.Models;
+using GraphQL;
+using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.Mvc.Infrastructure;
+using Microsoft.AspNetCore.Mvc.Routing;
+using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.Options;
+using OrchardCore.Autoroute.Models;
+using OrchardCore.ContentFields.Fields;
+using OrchardCore.ContentManagement;
+using OrchardCore.ContentManagement.Records;
+using OrchardCore.ContentManagement.Routing;
+using OrchardCore.Menu.Models;
+using System;
+using System.Threading.Tasks;
+using YesSql;
+
+namespace Etch.OrchardCore.Menu.Helpers
+{
+    public class CommonMenuItemPartMigrator : ICommonMenuItemPartMigrator
+    {
+        private readonly IContentManager _contentManager;
+        private readonly YesSql.ISession _session;
+        private readonly AutorouteOptions _autorouteOptions;
+        private readonly IUrlHelperFactory _urlHelperFactory;
+        private readonly IActionContextAccessor _actionContextAccessor;
+        private readonly IHttpContextAccessor _httpContextAccessor;
+        private readonly ILogger<CommonMenuItemPartMigrator> _logger;
+
+        public CommonMenuItemPartMigrator(IContentManager contentManager,
+            YesSql.ISession session,
+            IOptions<AutorouteOptions> autorouteOptions,
+            IUrlHelperFactory urlHelperFactory,
+            IActionContextAccessor actionContextAccessor,
+            IHttpContextAccessor httpContextAccessor,
+            ILogger<CommonMenuItemPartMigrator> logger)
+        {
+            _contentManager = contentManager;
+            _session = session;
+            _autorouteOptions = autorouteOptions.Value;
+            _urlHelperFactory = urlHelperFactory;
+            _actionContextAccessor = actionContextAccessor;
+            _httpContextAccessor = httpContextAccessor;
+            _logger = logger;
+        }
+
+        public async Task GetCommonMenuitemDataAsync()
+        {
+            var contentItems = await _session.Query<ContentItem>()
+                .With<ContentItemIndex>()
+                    .Where(x => x.Latest && x.ContentType == "Menu")
+                .ListAsync();
+
+            foreach (var contentItem in contentItems)
+            {
+                var menuItemsListPart = contentItem.As<MenuItemsListPart>();
+
+                foreach (var linkMenuItem in menuItemsListPart.MenuItems)

```

Comment:
I don't like how long this method is, let's break it up by moving the migration of menu items to it's own method that loops over the menu items and converts them.